DigitalLearningSolutions.Data/DataServices/CentresDataService.cs

```diff
@@ -584,5 +592,75 @@ public void ReactivateCentre(int centreId)
                 new { centreId }
             );
         }
+        public ContractInfo? GetContractInfo(int centreId)
+        {
+            var centre = connection.QueryFirstOrDefault<ContractInfo>(
+                @"SELECT c.CentreID,
+                            c.CentreName,
+                            ct.ContractTypeID,
+                            ct.ContractType,
+                            ct.ServerSpaceBytesInc,
+                            ct.DelegateUploadSpace,
+                             c.ContractReviewDate
+					    FROM Centres AS c
+                        INNER JOIN ContractTypes AS ct ON ct.ContractTypeID = c.ContractTypeId
+                        WHERE CentreID = @centreId",
+                new { centreId }
+            );
+
+            if (centre == null)
+            {
+                logger.LogWarning($"No centre found for centre id {centreId}");
+                return null;
+            }
+            return centre;
+        }
+        public bool UpdateContractTypeandCenter(
+           int centreId,
+           int contractTypeID,
+           long delegateUploadSpace,
+          long serverSpaceBytesInc,
+          DateTime? contractReviewDate
+       )
+        {
+            var numberOfAffectedRows = connection.Execute(
+                @" BEGIN TRY
+                    BEGIN TRANSACTION
+                        UPDATE ContractTypes SET

```

Comment:
we do not want to update the ContractTypes table here. We need to get the values from this table and update the Centres table.